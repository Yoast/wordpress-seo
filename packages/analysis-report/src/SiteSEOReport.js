import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

import { ScoreAssessments as SiteSEOReportAssessments, StackedProgressBar } from "@yoast/components";

/**
 * SeoAssessment container.
 */
const SiteSEOReportContainer = styled.div`
`;

/**
 * SeoAssessment top text.
 */
const SiteSEOReportText = styled.p`
	font-size: 14px;
`;

/**
 * The Dashboard Seo Assessment component.
 *
 * @param {object} props The component props.
 *
 * @returns {ReactElement} The react component.
 */
const SiteSEOReport = ( props ) => {
	return (
		<SiteSEOReportContainer
			className={ props.className }
		>
			<SiteSEOReportText
				className={ `${ props.className }__text` }
			>
				{ props.seoAssessmentText }
			</SiteSEOReportText>
			<StackedProgressBar
				className="progress"
				items={ props.seoAssessmentItems }
				barHeight={ props.barHeight }
			/>
			<SiteSEOReportAssessments
				className="assessments"
				items={ props.seoAssessmentItems }
			/>
		</SiteSEOReportContainer>
	);
};

SiteSEOReport.propTypes = {
	className: PropTypes.string,
	seoAssessmentText: PropTypes.string,
	seoAssessmentItems: PropTypes.arrayOf(
		PropTypes.shape( {
			html: PropTypes.string.isRequired,
			value: PropTypes.number.isRequired,
			color: PropTypes.string.isRequired,
		} )
	),
	barHeight: PropTypes.string,
};

SiteSEOReport.defaultProps = {
	className: "seo-assessment",
	seoAssessmentText: "SEO Assessment",
	seoAssessmentItems: null,
	barHeight: "24px",
};

export default SiteSEOReport;
