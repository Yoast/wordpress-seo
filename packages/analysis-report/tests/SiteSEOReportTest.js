import React from "react";
import renderer from "react-test-renderer";

import SiteSEOReport from "../src/SiteSEOReport";

describe( "The SiteSEOReport testing suite", () => {
	const siteSEOReportItems = [
		{
			html: "<div></div>",
			value: 10,
			color: "blue",
		},
		{
			html: "<div></div>",
			value: 15,
			color: "red",
		},
		{
			html: "<div></div>",
			value: 20,
			color: "yellow",
		},
	];

	const component = renderer.create(
		<SiteSEOReport
			className="test_SiteSEOReport"
			seoAssessmentText="Test assessment"
			seoAssessmentItems={ siteSEOReportItems }
			barHeight={ "10" }
		/>
	);

	const tree = component.toJSON();
	it( "The SiteSEOReport should match the snapshot", function() {
		expect( tree ).toMatchSnapshot();
	} );
} );
