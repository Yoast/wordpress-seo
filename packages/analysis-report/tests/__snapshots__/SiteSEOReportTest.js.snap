// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The SiteSEOReport testing suite The SiteSEOReport should match the snapshot 1`] = `
<div
  className="SiteSEOReport__SiteSEOReportContainer-yh1o4t-0 test_SiteSEOReport"
>
  <p
    className="SiteSEOReport__SiteSEOReportText-yh1o4t-1 hlEmAn test_SiteSEOReport__text"
  >
    Test assessment
  </p>
  <div
    className="StackedProgressBar__StackedProgressBarContainer-sc-10t3bky-0 eFtDnv progress"
  >
    <span
      className="StackedProgressBar__StackedProgressBarProgress-sc-10t3bky-1 pPvst progress__part"
    />
    <span
      className="StackedProgressBar__StackedProgressBarProgress-sc-10t3bky-1 kIFhAU progress__part"
    />
    <span
      className="StackedProgressBar__StackedProgressBarProgress-sc-10t3bky-1 bXkyX progress__part"
    />
  </div>
  <ul
    className="ScoreAssessments__ScoreAssessmentList-sc-163g4zx-5 iTRYwb assessments"
    role="list"
  >
    <li
      className="ScoreAssessments__ScoreAssessmentItem-sc-163g4zx-0 gshrRU assessments__item"
    >
      <span
        className="ScoreAssessments__ScoreAssessmentPart-sc-163g4zx-1 ScoreAssessments__ScoreAssessmentBullet-sc-163g4zx-2 hHvnje dFqjzq assessments__item-bullet"
      />
      <span
        className="ScoreAssessments__ScoreAssessmentPart-sc-163g4zx-1 ScoreAssessments__ScoreAssessmentText-sc-163g4zx-3 hHvnje dsIkPT assessments__item-text"
        dangerouslySetInnerHTML={
          Object {
            "__html": "<div></div>",
          }
        }
      />
      <span
        className="ScoreAssessments__ScoreAssessmentPart-sc-163g4zx-1 ScoreAssessments__ScoreAssessmentScore-sc-163g4zx-4 hHvnje cKSbYs assessments__item-score"
      >
        10
      </span>
    </li>
    <li
      className="ScoreAssessments__ScoreAssessmentItem-sc-163g4zx-0 gshrRU assessments__item"
    >
      <span
        className="ScoreAssessments__ScoreAssessmentPart-sc-163g4zx-1 ScoreAssessments__ScoreAssessmentBullet-sc-163g4zx-2 hHvnje hpnNZT assessments__item-bullet"
      />
      <span
        className="ScoreAssessments__ScoreAssessmentPart-sc-163g4zx-1 ScoreAssessments__ScoreAssessmentText-sc-163g4zx-3 hHvnje dsIkPT assessments__item-text"
        dangerouslySetInnerHTML={
          Object {
            "__html": "<div></div>",
          }
        }
      />
      <span
        className="ScoreAssessments__ScoreAssessmentPart-sc-163g4zx-1 ScoreAssessments__ScoreAssessmentScore-sc-163g4zx-4 hHvnje cKSbYs assessments__item-score"
      >
        15
      </span>
    </li>
    <li
      className="ScoreAssessments__ScoreAssessmentItem-sc-163g4zx-0 gshrRU assessments__item"
    >
      <span
        className="ScoreAssessments__ScoreAssessmentPart-sc-163g4zx-1 ScoreAssessments__ScoreAssessmentBullet-sc-163g4zx-2 hHvnje jPwpmU assessments__item-bullet"
      />
      <span
        className="ScoreAssessments__ScoreAssessmentPart-sc-163g4zx-1 ScoreAssessments__ScoreAssessmentText-sc-163g4zx-3 hHvnje dsIkPT assessments__item-text"
        dangerouslySetInnerHTML={
          Object {
            "__html": "<div></div>",
          }
        }
      />
      <span
        className="ScoreAssessments__ScoreAssessmentPart-sc-163g4zx-1 ScoreAssessments__ScoreAssessmentScore-sc-163g4zx-4 hHvnje cKSbYs assessments__item-score"
      >
        20
      </span>
    </li>
  </ul>
</div>
`;
